signer/src/block_observer.rs

```diff
@@ -436,11 +440,16 @@ impl<C: Context, B> BlockObserver<C, B> {
                 continue;
             }
 
+            let txid = tx.compute_txid();
```

Comment:
QQ: What's the actual downside/risk of having coinbase transactions extracted/stored when the deposit requests themselves will fail validation? I'm just pondering over what we gain by giving them special treatment vs. other failed deposits (your test would have been much simpler too).

Reply:
We wouldn't extract deposits txs regardless (since they are not spent "to the signers" directly), so this is only relevant for donations. In that case, I guess the main issue is that they cannot be spent right away -- this is not so relevant now that the system is up and running, but for the same reason it doesn't seems we are gaining anything by supporting them. OTOH, having them in the mix could mess up some assumptions on "any tx we care about have prevouts" (eg, we check that to distinguish between donations and sbtc tx)

Reply:
Yeah, the diff is about as small, I'm checking whether it will work now.

Reply:
Ah right, was thinking of the coinbase as a deposit but of course that's not the case 🤦‍♂️

I'm fine with the check, it's tiny (unlike the test to verify it...), my question was primarily out of a trying-to-minimize-special-case-handling perspective.

Reply:
I'll leave this open since apparently daniel is checking something, but feel free to close it.

Reply:
I just realized now that I was probably responding to the wrong thing above. Whoops!